### PR TITLE
frontend/fix/4xx-driver-arrived-at-pickup-api

### PR DESCRIPTION
**Pull request by Anuragini-2506**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

From Frontend the distance calculation between the current location and the pickup location for driver is less than 50 m, that's why the arrived/pickup API is called.
However, while sending the current lat long in the request body of the arrived/pickup API, there is a possibilty the lat long are the old one. ( due to the use of the  `getLastKnownLocValue ST.LATITUDE lat` and `getLastKnownLocValue ST.LONGITUDE lng` multiple times in the flow of calling arrived/pickup API ).

Fix - get the value only once and use it throughout the flow of calling arrived/pickup API to avoid this case.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 1]
```diff
@@ -750,15 +751,19 @@ eval RetryTimeUpdate state = do
   (updateAndExit state { data = state.data { locationLastUpdatedTime = "" }, props = state.props {refreshAnimation = true}} $ Refresh state { data = state.data { locationLastUpdatedTime = "" }, props = state.props {refreshAnimation = true}})
 
 eval (TimeUpdate time lat lng) state = do
-  let newState = state { data = state.data { currentDriverLat= getLastKnownLocValue ST.LATITUDE lat,  currentDriverLon = getLastKnownLocValue ST.LONGITUDE lng, locationLastUpdatedTime = (convertUTCtoISC time "hh:mm a") }}
+  let driverLat = getLastKnownLocValue ST.LATITUDE lat
+      driverLong = getLastKnownLocValue ST.LONGITUDE lng
+      newState = state { data = state.data { currentDriverLat= driverLat,  currentDriverLon = driverLong, locationLastUpdatedTime = (convertUTCtoISC time "hh:mm a") }}
+      hello = spy "Hello world.. today was a bad day" (show driverLat <> " " <> show driverLong)
```
Comment (aryan-13) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:757]: Remove logs

**Review by aryan-13 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 2]
```diff
@@ -750,15 +751,18 @@ eval RetryTimeUpdate state = do
...
   void $ pure $ setValueToLocalStore LOCATION_UPDATE_TIME (convertUTCtoISC time "hh:mm a")
   continueWithCmd newState [ do
     void $ if (getValueToLocalNativeStore IS_RIDE_ACTIVE == "false") then checkPermissionAndUpdateDriverMarker newState else pure unit
     case state.data.config.waitTimeConfig.enableWaitTime, state.props.currentStage, state.data.activeRide.notifiedCustomer, state.data.snappedOrigin of
-      true, ST.RideAccepted, false, Nothing -> pure (UpdateOriginDist lat lng)
+      true, ST.RideAccepted, false, Nothing -> pure (UpdateOriginDist driverLat driverLong)
       true, ST.RideAccepted, false, Just snapped -> do
-        let dist = (getDistanceBwCordinates (getLastKnownLocValue ST.LATITUDE lat) (getLastKnownLocValue ST.LONGITUDE lng)  snapped.lat snapped.lon)
+
+        let dist = (getDistanceBwCordinates (driverLat) (driverLong)  snapped.lat snapped.lon)
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:765]: Remove extra brackets

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 3]
```diff
@@ -774,7 +778,7 @@ eval (UpdateOriginDist lat lng) state =
         Right (API.RideRouteResp resp) -> 
           case Array.head resp.points of
             Just (API.LatLong loc) -> do
-              let dist = getDistanceBwCordinates (getLastKnownLocValue ST.LATITUDE lat) (getLastKnownLocValue ST.LONGITUDE lng) loc.lat loc.lon
+              let dist = getDistanceBwCordinates (lat) (lng) loc.lat loc.lon
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:781]: Remove extra brackets

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs (hunk 2 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs]: done

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs (hunk 3 above)
Comment (Anuragini-2506) [reply Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs]: done

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -87,6 +87,7 @@ import Engineering.Helpers.Commons (liftFlow)
 import PrestoDOM.Core (getPushFn)
 import Control.Monad.Except.Trans (lift)
 import Data.Either (Either(..))
+import Debug (spy)
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:90]: remove redundant imports

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**